core/src/backend/ui.rs

```diff
@@ -1,5 +1,51 @@
 use crate::events::{KeyCode, PlayerEvent};
+pub use crate::loader::Error as LoaderError;
+use chrono::{DateTime, Utc};
+use downcast_rs::Downcast;
 use std::collections::HashSet;
+use std::future::Future;
+use std::pin::Pin;
+
+/// Type alias for pinned, boxed, and owned futures that output a falliable
+/// result of type `Result<T, E>`.
+pub type OwnedFuture<T, E> = Pin<Box<dyn Future<Output = Result<T, E>> + 'static>>;
```

Comment:
Looks like this was taken as-is from `navigator.rs`. Why not importing it?

Reply:
Not sure why this was done, but it's changed now